README.md

```diff
@@ -1,43 +1,128 @@
+This will add or remove brokers to match the required number. To scale a deployment with 3 kafka units to 5, run:
+```shell
+juju scale-application kafka-k8s 5
+```
 
+### Password rotation
+#### Internal operator user
+The operator user is used internally by the Charmed MongoDB Operator, the `set-password` action can be used to rotate its password.
+```shell
+# to set a specific password for the operator user
+juju run-action kafka-k8s/leader set-password password=<password> --wait
 
-### Checklist
+# to randomly generate a password for the operator user
+juju run-action kafka-k8s/leader set-password --wait
+```
 
-- [x] Super-user creation
-- [x] Inter-broker auth
-- [x] Horizontally scale brokers
-- [x] Username/Password creation for related applications
-- [ ] Automatic topic creation with associated user ACLs
-- [ ] Partition rebalancing during broker scaling
-- [ ] Rack awareness support
-- [ ] Persistent storage support with [Juju Storage](https://juju.is/docs/olm/defining-and-using-persistent-storage)
-- [ ] TLS/SSL support
+## Relations
+
+Supported [relations](https://juju.is/docs/olm/relations):
+
+#### `tls` interface:
+
+To enable TLS:
+
+```shell
+# deploy the TLS charm 
+juju deploy tls-certificates-operator --channel=edge
+# add the necessary configurations for TLS
+juju config tls-certificates-operator generate-self-signed-certificates="true" ca-common-name="Test CA" 
+# to enable TLS relate the two applications 
+juju relate tls-certificates-operator zookeeper-k8s
+juju relate tls-certificates-operator kafka-k8s
+```
+
+Updates to private keys for certificate signing requests (CSR) can be made via the `set-tls-private-key` action.
+```shell
+# Updates can be done with auto-generated keys with
+juju run-action kafka-k8s/0 set-tls-private-key --wait
+juju run-action kafka-k8s/1 set-tls-private-key --wait
+juju run-action kafka-k8s/2 set-tls-private-key --wait
+```
+
+Passing keys to external/internal keys should *only be done with* `base64 -w0` *not* `cat`. With three brokers this schema should be followed:
+```shell
+# generate shared internal key
+openssl genrsa -out internal-key.pem 3072
+# generate external keys for each unit
+openssl genrsa -out external-key-0.pem 3072
+openssl genrsa -out external-key-1.pem 3072
+openssl genrsa -out external-key-2.pem 3072
+# apply both private keys on each unit, shared internal key will be allied only on juju leader
+juju run-action kafka-k8s/0 set-tls-private-key "external-key=$(base64 -w0 external-key-0.pem)"  "internal-key=$(base64 -w0 internal-key.pem)"  --wait
+juju run-action kafka-k8s/1 set-tls-private-key "external-key=$(base64 -w0 external-key-1.pem)"  "internal-key=$(base64 -w0 internal-key.pem)"  --wait
+juju run-action kafka-k8s/2 set-tls-private-key "external-key=$(base64 -w0 external-key-2.pem)"  "internal-key=$(base64 -w0 internal-key.pem)"  --wait
```

Comment:
I'm not sure we have the `external-key` option in that action. I believe we have to adapt the Config section as well